e2e/cypress/integration/files_and_attachments/image_link_preview_spec.js

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Image Link Preview', () => {
+    let testTeam;
+
+    before(() => {
+        // # Create new team and new user and visit Town Square channel
+        cy.apiInitSetup({loginAfter: true}).then(({team}) => {
+            testTeam = team;
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+        });
+    });
+
+    it('MM-T331 Image link preview - Collapse and expand', () => {
+        const link = 'http://www.traveller.com.au/content/dam/images/g/u/n/q/h/0/image.related.articleLeadwide.620x349.gunpvd.png/1488330286332.png';
+
+        const expanded = 'button[data-expanded="true"]';
+        const collapsed = 'button[data-expanded="false"]';
+
+        // # Ensure Account Settings -> Display -> Default appearance of image previews is set to Expanded
+
+        cy.get('#sidebarHeaderDropdownButton').should('be.visible').click();
+
+        cy.get('.dropdown-menu').should('be.visible').findByText('Account Settings').click();
+
+        cy.get('.settings-links').should('be.visible').findByText('Display').click();
+
+        cy.get('#collapseDesc').within(() => {
+            cy.get('span').contains('Expanded');
+        });
+
+        cy.get('#accountSettingsHeader > .close').click();
+
+        // # Post a link to an externally hosted image
+        cy.postMessage(link);
+
+        // # Click to reply to that post, and post that same link again (so you can see it twice in both center and RHS)
+        cy.clickPostCommentIcon();
+
+        cy.postMessageReplyInRHS(link);
+
+        cy.getLastPostId().then((postId) => {
+            cy.get(`#post_${postId}`).within(() => {
+                cy.get(expanded).click().then(($element) => {
+                    expect($element).to.have.attr('data-expanded', 'false');
+                });
+            });
```

Comment:
This seemed to be not meeting the minimum requirement of:
  3. In center, on one of the posts, click the collapse arrows to collapse the image preview
  4. Observe it collapses in both center and RHS view

Also, it should verify initial render state

Could be something like
```suggestion
            // * Verify it renders correctly both on center and RHS view
            cy.get(`#post_${postId}`).should('be.visible').within(() => {
                cy.findByLabelText('Toggle Embed Visibility').
                    should('be.visible').and('have.attr', 'data-expanded', 'true');
                cy.findByLabelText('file thumbnail').should('be.visible');
            });
            cy.get(`#rhsPost_${postId}`).should('be.visible').within(() => {
                cy.findByLabelText('Toggle Embed Visibility').
                    should('be.visible').and('have.attr', 'data-expanded', 'true');
                cy.findByLabelText('file thumbnail').should('be.visible');
            });

            // # In center, click the collapse arrows to collapse the image preview
            cy.get(`#post_${postId}`).findByLabelText('Toggle Embed Visibility').
                click().
                should('have.attr', 'data-expanded', 'false');

            // * Observe it collapses in both center and RHS view
            cy.get(`#post_${postId}`).findByLabelText('file thumbnail').should('not.exist');
            cy.get(`#rhsPost_${postId}`).findByLabelText('file thumbnail').should('not.exist');
```